### PR TITLE
[MRG] Rnn, gru,lstm Blocks

### PR DESCRIPTION
**Pull request by PraveenVenugopal**
### Which issue(s) does this Pull Request fix?
Issue #652 

### Pull Request Status 
The pull request is ready to be merged. 

### Details of the Pull Request
RNN, GRU, and LSTM have been implemented as separate blocks in hyper_block.py. All three blocks are multi-layered and bi-directional. The number of layers is chosen via hp.Choice() ie, it's a configurable hyperparameter

### CODE REVIEW

**Comment by PraveenVenugopal** [comment]
Sure. Will do them

File: tests/test_hyper_block.py [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 from autokeras.auto.auto_model import *
 from autokeras import const
-from autokeras.hypermodel.hyper_block import XceptionBlock
-from autokeras.hypermodel.hyper_head import ClassificationHead
+
+from autokeras.hypermodel.hyper_block import XceptionBlock, RNNBlock
```
Comment (jhfjhfj1) [inline tests/test_hyper_block.py]: To follow the keras style guide, the imports should be only importing modules and one module per line.

`from autokeras.hypermodel import hyper_block
from autokeras.hypermodel import hyper_head
...
`

Comment (PraveenVenugopal) [reply tests/test_hyper_block.py]: Similar import statements are present in hyper_block.py as well. Fixing them too.

Just to clarify, do you really prefer imports to be like :
`from autokeras.hypermodel import hyper_block`

or is it fine if I do like below
`from autokeras.hypermodel.hyper_block import *`

In first case, I must replace all occurrences with _hyper_block.XceptionBlock_  or _hyper_block.RNNBlock_. 
The second case doesn't require that (changes will happen only to import statements)

File: autokeras/hypermodel/hyper_block.py [hunk 2]
```diff
@@ -50,7 +50,35 @@ def build(self, hp, inputs=None):
...
+        type_of_block = hp.Choice('rnn_type', ['vanilla', 'gru', 'lstm'], default='vanilla')
+        if type_of_block == 'vanilla':
+            print("make vanilla")
+            in_layer = tf.keras.layers.SimpleRNN
+        elif type_of_block == 'gru':
+            print("make gru")
+            in_layer = tf.keras.layers.GRU
+        elif type_of_block == 'lstm':
+            print("make lstm")
+            in_layer = tf.keras.layers.LSTM
```
Comment (tonyabracadabra) [inline autokeras/hypermodel/hyper_block.py]: I would wrap it in a dictionary for simplicity

File: autokeras/hypermodel/hyper_block.py [hunk 3]
```diff
@@ -50,7 +50,35 @@ def build(self, hp, inputs=None):
...
+        choice_of_layers = hp.Choice('num_layers', [1, 2, 3], default=2)
+        for i in range(choice_of_layers):
+            if i == choice_of_layers - 1:
+                bidirectional_block = tf.keras.layers.Bidirectional(
+                    in_layer(layer_units, activation='tanh', use_bias=True,
+                                              return_sequences=False))
+            else:
+                bidirectional_block = tf.keras.layers.Bidirectional(
+                    in_layer(layer_units, activation='tanh', use_bias=True,
+                                              return_sequences=True))
```
Comment (tonyabracadabra) [inline autokeras/hypermodel/hyper_block.py]: I would use 
```
return_sequences = False if i == choice_of_layers - 1 else True
bidirectional_block = tf.keras.layers.Bidirectional(
                    in_layer(layer_units, activation='tanh', use_bias=True,
                                              return_sequences=return_sequences))
```
for readability

**Pull request merged**